lib/mqtt.c

```diff
@@ -737,6 +749,35 @@ static CURLcode mqtt_done(struct Curl_easy *data,
   return CURLE_OK;
 }
 
+/* we ping every 30 seconds to avoid being disconnected by the server */
+static CURLcode mqtt_ping(struct Curl_easy *data)
+{
+  CURLcode result = CURLE_OK;
+  struct connectdata *conn = data->conn;
+  struct mqtt_conn *mqtt = &conn->proto.mqtt;
+  struct MQTT *mq = data->req.p.mqtt;
+
+  if(mqtt->state == MQTT_FIRST &&
+     mq->pingsent == 0 &&
+     data->set.tcp_keepalive) {
+    struct curltime t = Curl_now();
+    timediff_t diff = Curl_timediff(t, mq->lastTime);
+
+    if(diff > 30000) {
```

Comment:
How about using a define for the time to make it easier to adjust and avoid the magic number in the code?

Reply:
Good idea.

Reply:
And changed.